modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/slot/SlotManagerClusterMode.java

```diff
@@ -296,37 +264,22 @@ private Slot getOverlappedSlot(String nodeId, String queueName) {
      * @param nodeId        ID of the node to which slot is Assigned
      */
     private void updateSlotAssignmentMap(String queueName, Slot allocatedSlot, String nodeId) {
-        TreeSet<Slot> currentSlotList;
-        HashMap<String, TreeSet<Slot>> queueToSlotMap;
 
         //Lock is used because this method will be called by multiple nodes at the same time
         String lockKey = nodeId + SlotManagerClusterMode.class;
         synchronized (lockKey.intern()) {
 
-            HashmapStringTreeSetWrapper wrapper = slotAssignmentMap.get(nodeId);
-            if (wrapper == null) {
-                wrapper = new HashmapStringTreeSetWrapper();
-                queueToSlotMap = new HashMap<String, TreeSet<Slot>>();
-                wrapper.setStringListHashMap(queueToSlotMap);
-                slotAssignmentMap.putIfAbsent(nodeId, wrapper);
+            /**
+             * Update assigned node, assigned queue and set state to assigned
+             */
+            try {
+                rdbmsSlotAgent
+                        .updateSlotAssignment(nodeId, queueName, allocatedSlot.getStartMessageId(),
+                                              allocatedSlot.getEndMessageId());
+            } catch (Exception ex) {
+                log.error("Failed to assign " + allocatedSlot.getSlotId() + " for queue " + queueName);
```

Comment:
Can you explain why we are not throwing this exception here? Would be great if you can log the exception as well here. You are ignoring the stack trace here.